feathr_project/registry/registry/models.py

```diff
@@ -0,0 +1,707 @@
+            EntityType.AnchorFeature: "feathr_anchor_feature_v1",
+            EntityType.DerivedFeature: "feathr_derived_feature_v1",
+        }[self]
+
+
+class RelationshipType(Enum):
+    Contains = 1
+    BelongsTo = 2
+    Consumes = 3
+    Produces = 4
+
+
+class ToDict(ABC):
+    @abstractmethod
+    def to_dict(self) -> dict:
+        pass
+
+    def to_json(self, indent=None) -> str:
+        return json.dumps(self.to_dict(), indent=indent)
+
+
+class FeatureType(ToDict):
+    def __init__(self,
+                 type: Union[str, VectorType],
+                 tensor_category: Union[str, TensorCategory],
+                 dimension_type: list[Union[str, ValueType]],
+                 val_type: Union[str, ValueType]):
+        self.type = to_type(type, VectorType)
+        self.tensor_category = to_type(tensor_category, TensorCategory)
+        self.dimension_type = to_type(dimension_type, ValueType)
+        self.val_type = to_type(val_type, ValueType)
+
+    def to_dict(self) -> dict:
+        return {
+            "type": self.type.name,
+            "tensorCategory": self.tensor_category.name,
+            "dimensionType": [t.name for t in self.dimension_type],
+            "valType": self.val_type.name,
+        }
+
+
+class TypedKey(ToDict):
+    def __init__(self,
+                 key_column: str,
+                 key_column_type: ValueType,
+                 full_name: Optional[str] = None,
+                 description: Optional[str] = None,
+                 key_column_alias: Optional[str] = None):
+        self.key_column = key_column
+        self.key_column_type = to_type(key_column_type, ValueType)
+        self.full_name = full_name
+        self.description = description
+        self.key_column_alias = key_column_alias
+
+    def to_dict(self) -> dict:
+        ret = {
+            "key_column": self.key_column,
+            "key_column_type": self.key_column_type.name,
+        }
+        if self.full_name is not None:
+            ret["full_name"] = self.full_name
+        if self.description is not None:
+            ret["description"] = self.full_name
+        if self.key_column_alias is not None:
+            ret["key_column_alias"] = self.key_column_alias
+        return ret
+
+
+class Transformation(ToDict):
+    @staticmethod
+    def new(**kwargs):
+        if "transform_expr" in kwargs:
+            return ExpressionTransformation(**kwargs)
+        elif "def_expr" in kwargs:
+            return WindowAggregationTransformation(**kwargs)
+        elif "name" in kwargs:
+            return UdfTransformation(**kwargs)
+        else:
+            raise ValueError(kwargs)
+
+
+class ExpressionTransformation(Transformation):
+    def __init__(self, transform_expr: str):
+        self.transform_expr = transform_expr
+
+    def to_dict(self) -> dict:
+        return {
+            "transform_expr": self.transform_expr
+        }
+
+
+class WindowAggregationTransformation(Transformation):
+    def __init__(self,
```

Comment:
We should avoid duplicated feature definition code. Maybe extend the existing class?